### ISSUE TITLE
ITV Plugin - "sometimes" reports no playable streams found 

### ISSUE THREAD

**Issue by lcsneil** [streamlink/streamlink#1436]
- [x] This is a bug report.
- [ ] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [ x] I have read the contribution guidelines.

Description
ITV plugin reports no playable stream found - but only for some programmes (not always the newest ones either). 
### Reproduction steps / Explicit stream URLs to test

1. Plugin works on this URL https://www.itv.com/hub/minder/685111
2. Plugin DOESNT work on this URL https://www.itv.com/hub/minder/685112 (note that it does play using a web broswer)

### Environment details

Win10 Pro 32 bit, v1709, 
Streamlink (0.9.0) and Python version (3.5.2) 

Thanks
lcsneil

**Comment by beardypig** [comment]
Seems to be working for me, right now at least.

**Comment by beardypig** [comment]
Looks like they are introducing a new API - we'll have to update the plugin to support it :)

**Comment by lcsneil** [comment]
For info the Christmas edition of Victoria gave the same 'no stream found' issue,

https://www.itv.com/hub/victoria/2a4229a0017

lcsneil

**Comment by lcsneil** [comment]
Now seem to be getting a different error on the ITV plugin (have upgraded to 0.10.0)

error: Unable to open URL: http://www.itv.com/hub/marcella/2a4269a0009 (HTTPSConnectionPool(host='www.itv.com', port=443): Max retries exceeded with url: /hub/marcella/2a4269a0009 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:645)'),)))

Win10 Pro 32 bit, v1709,
Streamlink (0.9.0) and Python version (3.5.2)

**Comment by Vangelis66** [comment]
> error: Unable to open URL: http://www.itv.com/hub/marcella/2a4269a0009

... Have you tried to use the default HTTPS secure link from the start: 

https://www.itv.com/hub/marcella/2a4269a0009

rather than use the plain HTTP one (which invokes auto redirection to HTTPS all the same) ?

**Comment by lcsneil** [comment]
No I hadn't but gives the same result:-

error: Unable to open URL: https://www.itv.com/hub/marcella/2a4269a0009/ (HTTPSConnectionPool(host='www.itv.com', port=443): Max retries exceeded with url: /hub/marcella/2a4269a0009/ (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:645)'),)))

**Comment by Vangelis66** [comment]
@lcsneil 

AFAICT, there's no issue with streamlink's ITVplayer plugin and the referenced ITVhub link! :relieved:

**OS:** Windows Vista SP2 32-bit, with post EOL MS updates (meant for WinServer 2008 SP2)
**Streamlink version**: The _most excellent_ pre-compiled portable package prepared by @RosadinTV, available from [here](https://github.com/streamlink/streamlink-portable/releases/tag/Stable_v0.10.0); this uses Python 3.6.3 (x86) and release version 0.10.0 of the streamlink script.

As I'm not inside the UK, I used a known trick :wink: in my command to access the geo-fenced playlist, while the actual RTMPE stream is fetched DIRECT (as the FMS server does not geo-block); the exact command I issued is: 
`streamlink -l debug --http-header X-Forwarded-For=31.3.228.238 -o "Marcella_SE2E01[2a4269a0009].flv" "https://www.itv.com/hub/marcella/2a4269a0009" best`
Result: 
```
[Streamlink for Windows v0.10.0 - Git b839cfd]
[cli][info] Found matching plugin itvplayer for URL https://www.itv.com/hub/marc
ella/2a4269a0009
[cli][info] Available streams: 600k (worst), 800k, 1200k, 1500k, 1800k (best)
[cli][info] Opening stream: 1800k (rtmp)
[stream.][debug] Spawning command: G:\Applications\Streamlink\StreamlinkPortable
[byRosadinTV]\Streamlink-portable-0.10.0-py3.6.3-win32\Streamlink\Dependencies\r
tmpdump\rtmpdump.exe --app ondemand?auth=eaEd7dVcYaLaqblatcRaFcJaubFbKcIaSbab8aJ
d2csbDarcXcHcjaad7b7cqdDcka4-bAKEQ4-eS-dAt3CAujBsxp&aifp=v002&slist=production/
--flv - --playpath mp4:production/priority/CATCHUP/2/4269/0009/001/2-4269-0009-0
01_PC011800_16X9.mp4 --rtmp rtmpe://cp72511.edgefcs.net/ondemand --swfVfy http:/
/www.itv.com/mercury/Mercury_VideoPlayer.swf --timeout 10
[cli][debug] Pre-buffering 8192 bytes
[cli][debug] Checking file output
[cli][debug] Writing stream to output
[download][..SE2E01[2a4269a0009].flv] Written 504.3 MB (6m59s @ 1.2 MB/s)
[cli][info] Stream ended
[cli][info] Closing currently open stream...
[End of Streamlink for Windows with ExitCode 0]
```
... If you're also overseas, me thinks your issue is caused by your proxy/VPN/VPS/SmartDNS etc... :stuck_out_tongue: ; BTW, you didn't inform us of your location (I suppose it's a required piece of info for geo-sensitive plugins); if in the UK, then something's awry with your secure connection to the ITVhub host...

**Comment by VladiDamian** [comment]
Hi @lcsneil 
Check that the date on your computer are set correctly.
Also you can use the **--http-no-ssl-verify** to skip SSL verification although is not recommended (https://streamlink.github.io/cli.html#cmdoption-http-no-ssl-verify)

**Comment by RosadinTV** [comment]
Hi @lcsneil
Check that the date on your computer are set correctly.
Also you can use the --http-no-ssl-verify argument to skip SSL verification although is not recommended (https://streamlink.github.io/cli.html#cmdoption-http-no-ssl-verify)

**Comment by Vangelis66** [comment]
@RosadinTV 

With greatest respect, this issue **ISN'T SOLVED**, hence it shouldn't have been closed!
If you read (again) the very first post of this thread, you'll realise it's a case of what I described/explained further down [here](https://github.com/streamlink/streamlink/issues/1436#issuecomment-357735759)... 
**TL;DR**: 
Not all ITVhub shows (Video-On-Demand) are available as RTMPE streams (compatible with streamlink's itvplayer plugin); but all (including those that are failing with SL) are now available as AppleHLS streams (that the current iteration of the plugin can't intercept). 

 In addition to the above for VOD, ITV are also deprecating their **live** RTMPE streams for the group of their channels (currently only one variant @1200k available for each of them - used to be five, the best @1800k). 
Recent tests on desktop Firefox reveal that only ITV1+ITV4 (live) still employ Flash and RTMPE streams, while now ITV2+ITV3 live streams are of the encrypted MPEG-DASH type (with Widevine CDM used), delivered over HTML5... 

If you ask me, this issue should be closed only when the [new rewrite](https://github.com/streamlink/streamlink/issues/1436#issuecomment-357651466) of the plugin shows up, an updated version that deals with the API changes gradually put in place by the ITVhub people... So, kindly reconsider and reopen, thanks!
